etc/scripts/timetable/import.js

```diff
@@ -193,14 +210,231 @@ var _importCourses = function(ctx, courses, callback) {
         if (err) {
             log().error({'err': err}, 'Failed to import a course');
-            process.exit();
+            process.exit(1);
         }
 
         // Move on to the next course
         _importCourses(ctx, courses, callback);
     });
 };
+
+/**
+ * Export a set of courses. This method will add an `exportedCourse` on each course
+ * holding the entire tree for that course
+ *
+ * @param  {Context}    ctx         Standard context object containing the current user and the current application
+ * @param  {Object[]}   courses     The courses to export
+ * @param  {Function}   callback    Standard callback function
+ * @api private
+ */
+var _exportCourses = function(ctx, courses, callback) {
+    if (_.isEmpty(courses)) {
+        return callback();
+    }
+
+    var course = courses.pop();
+
+    // Export the course so the new ids can be used for the borrowing stage
+    OrgUnitAPI.exportOrgUnit(ctx, course.orgUnitId, 'json', function(err, exportedCourse) {
+        if (err) {
+            log().error({'err': err}, 'Failed to export a course');
+            process.exit(1);
+        }
+
+        course.exportedCourse = exportedCourse;
+
+        // Move on to the next course
+        _exportCourses(ctx, courses, callback);
+    });
+};
+
+/**
+ * Borrow series under a module where appropriate
+ *
+ * @param  {Context}    ctx         Standard context object containing the current user and the current application
+ * @param  {Object[]}   courses     The courses to check for borrowing
+ * @param  {Function}   callback    Standard callback function
+ * @api private
+ */
+var _handleBorrowing = function(ctx, courses, callback) {
+    // Get all the organisational units with borrowed series
+    var orgunitsWithBorrowedSeries = [];
+    _.each(courses, function(course) {
+        _getOrgUnitsWithBorrowedSeries(course, orgunitsWithBorrowedSeries);
+    });
+
+    // If there is not a single organisational unit with a series, we're done
+    if (_.isEmpty(orgunitsWithBorrowedSeries)) {
+        return callback();
+    }
+
+    // Otherwise, we need to process them
+    return _handleOrgUnitsWithBorrowedSeries(ctx, courses, orgunitsWithBorrowedSeries, callback);
+};
+
+/**
+ * Get the organisational units (modules) who contain series that are borrowed from
+ * other modules
+ *
+ * @param  {OrgUnit}        orgUnit                         The organisational unit to check
+ * @param  {OrgUnit[]}      orgunitsWithBorrowedSeries      The organisational units that contain borrowed series
+ */
+var _getOrgUnitsWithBorrowedSeries = function(orgUnit, orgunitsWithBorrowedSeries) {
+    if (!_.isEmpty(orgUnit.borrowedSeries)) {
+        orgunitsWithBorrowedSeries.push(orgUnit);
+    }
+
+    // Visit any child organisational units
+    _.each(orgUnit.children, function(childOrgUnit) {
+        _getOrgUnitsWithBorrowedSeries(childOrgUnit, orgunitsWithBorrowedSeries);
+    });
```

Comment:
This isn't actually getting and returning anything. I'd prefer if we can avoid having function contracts that are based on mutating arrays/objects